contracts/mocks/StakingPool/SPMockCover.sol

```diff
@@ -56,18 +62,22 @@ contract SPMockCover {
   ) public returns (uint premium) {
 
     Product memory product = products[params.productId];
-    uint gracePeriod = uint(productTypes[product.productType].gracePeriodInDays) * 1 days;
+    uint gracePeriod = productTypes[product.productType].gracePeriod;
 
-    return _stakingPool.allocateCapacity(
+    return _stakingPool.requestAllocation(
+      params.amount,
+      // TODO: figure out if these need to be populated
+      0, // previousPremium
       AllocationRequest(
         params.productId,
         coverId,
-        params.amount,
         params.period,
-        product.useFixedPrice
-      ),
-      AllocationRequestConfig(
         gracePeriod,
+        product.useFixedPrice,
+        // TODO: figure out if these need to be populated
```

Comment:
given that this is a test file, do we still need this TODO here?

Reply:
Yes, the field are relevant for the cover editing. We might need to set them while unit testing, hence the todo.